### PR TITLE
Eeg preprocessing

### PR DESCRIPTION
**Pull request by HermineBerberyan**

### CODE REVIEW

File: eeg_eyetracking_parser/_eeg_preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    return raw
+
+
+def band_pass_filter(raw, lf=0.1, hf=40, plot=False):
+    """
+    High-pass and low-pass filter for EEG data (band-pass filter)
+    High-pass removes slow drifts
+    Low-pass removes the short-term fluctuations
+    """
+    raw_unfiltered = raw.copy()
```
Comment (smathot) [inline eeg_eyetracking_parser/_eeg_preprocessing.py:76]: Only create unfiltered copy if it is actually used.

File: eeg_eyetracking_parser/_eeg_preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    raw.set_eeg_reference(ref_channels=ref_channels)
+    raw.drop_channels(ref_channels)
+    return raw
+
+
+def drop_unused_channels(raw, chan_name_pattern='Channel'):
+    """
+    Drop unused channels. The pattern is that they all start with "Channel"
+    """
+    empty_chan_names = [chan_name for chan_name in raw.info['ch_names'] if chan_name.startswith(chan_name_pattern)]
```
Comment (smathot) [inline eeg_eyetracking_parser/_eeg_preprocessing.py:24]: Avoid lines > 79 characters (here and elsewhere)

File: eeg_eyetracking_parser/_eeg_preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+import matplotlib.pyplot as plt
+
+
+def rereference_channels(raw, ref_channels=['A1', 'A2']):
+    """
+    Rereference EEG data to the mastoids
+    """
+    raw.set_eeg_reference(ref_channels=ref_channels)
+    raw.drop_channels(ref_channels)
+    return raw
```
Comment (smathot) [inline eeg_eyetracking_parser/_eeg_preprocessing.py:17]: Functions that modify the `raw` in place shouldn't return the `raw`. This behavior should be clearly described in the docstring.

File: eeg_eyetracking_parser/_eeg_preprocessing.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+import os
+from . import trial_trigger
+from mne.preprocessing import ICA
+import autoreject as ar
+import json
+import matplotlib.pyplot as plt
+
+
+def rereference_channels(raw, ref_channels=['A1', 'A2']):
+    """
```
Comment (smathot) [inline eeg_eyetracking_parser/_eeg_preprocessing.py:12]: Describe arguments in docstrings using numpydoc format (here and elsewhere). Also describe the behavior of the function, for example how it changes the `raw` object by modifying the signal or the `info` property.

File: eeg_eyetracking_parser/_eeg_preprocessing.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    raw_unfiltered = raw.copy()
+    raw.filter(l_freq=lf, h_freq=hf, picks='all')
+    if plot:
+        for title, data in zip(['Unfiltered', 'Bandpass filtered'], [raw_unfiltered, raw]):
+            fig = data.plot_psd(fmax=250)
+            fig.suptitle(title)
+    return raw
+
+
+def downsample_data(raw, srate=256):
```
Comment (smathot) [inline eeg_eyetracking_parser/_eeg_preprocessing.py:85]: Event timestamps should also be adjusted. (Due to the way that the EEG and eye-tracking data are merged, we cannot first downsample and then extract the events from the annotations, or at least not easily.)

**Review by smathot — CHANGES_REQUESTED** [review]
Thank you. A few requests for changes and then I will merge it. It seems to work, but because of the downsample-events issue, I haven't been able to properly test it, so I'll try this again after these changes.

File: eeg_eyetracking_parser/_eeg_preprocessing.py (hunk 1 above)
Comment (HermineBerberyan) [reply eeg_eyetracking_parser/_eeg_preprocessing.py:76]: It is used in plotting

**Pull request merged**